chrome/background.js

```diff
@@ -95,6 +95,42 @@ function onMessage(request, sender, sendResponse) {
       localStorage.getItem("use_fuzzy_search") != "false";
     sendResponse({ use_fuzzy_search: use_fuzzy_search });
   }
+
+  // spawn a new tab with pre-provided credentials
+  if (request.action == "launch") {
+    chrome.tabs.create({url: request.url}, function (tab) {
+      var authAttempted = false;
+      chrome.webRequest.onAuthRequired.addListener(
+        function authListener(requestDetails) {
+          // remove event listeners once tab loading is complete
+          chrome.tabs.onUpdated.addListener(function statusListener(tabId, info) {
+            if (info.status === "complete") {
+              chrome.tabs.onUpdated.removeListener(statusListener);
+              chrome.webRequest.onAuthRequired.removeListener(authListener);
+            }
+          });
+          // only supply credentials if this is the first time for this tab
+          if (!authAttempted) {
```

Comment:
1. Create a new tab, and get the tab ID in the callback
2. Add onAuthRequired listener
3. When the auth callback fires, add a status listener to the tab
4. Supply login credentials if we haven't already supplied them
5. When the status listener fires and the status is complete (i.e. the page has loaded), remove both listeners.

The reason both the `authCompleted` and the listeners are required:
 * The `onAuthRequired` listener cannot remove itself, as doing so crashes the extension.
 * Because it cannot remove itself, it may fire multiple times (e.g. if the password is incorrect)
 * So the boolean restricts it to one time only
 * And removing it in the status listener once the page is complete ensures that the event is no longer live, and avoids a crash.

> is it guaranteed that `info.status` will become complete only after basic auth is filled in?

I could not find any specific documentation guaranteeing this, and I looked pretty hard. However, the documentation does say that this occurs when the page load is complete, and blocking authentication requirements prevent the page from loading until they are satisfied. I've tested it extensively, and I'm comfortable that this does seem to be the case, but I cannot absolutely guarantee that it is so.

Reply:
thank you, makes sense!